internal/weldrcheck/source_test.go

```diff
@@ -0,0 +1,190 @@
+// Package weldrcheck - source contains functions to check the source API
+// Copyright (C) 2020 by Red Hat, Inc.
+
+// Tests should be self-contained and not depend on the state of the server
+// They should use their own blueprints, not the default blueprints
+// They should not assume version numbers for packages will match
+// They should run tests that depend on previous results from the same function
+// not from other functions.
+
+// +build integration
+
+package weldrcheck
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/osbuild/osbuild-composer/internal/client"
+)
+
+// POST a new JSON source
+func TestPOSTJSONSourceV0(t *testing.T) {
+	source := `{
+		"name": "package-repo-json-v0",
+		"url": "file://REPO-PATH",
+		"type": "yum-baseurl",
+		"proxy": "https://proxy-url/",
+		"check_ssl": true,
+		"check_gpg": true,
+		"gpgkey_urls": ["https://url/path/to/gpg-key"]
+	}`
+	source = strings.Replace(source, "REPO-PATH", testState.repoDir, 1)
```

Comment:
Is `testState` a global variable coming from another file ? I didn't know you can do this.

Reply:
Yes, all of this code is in the weldrcheck package, so it shares global variables. It is setup by the  TestMain function in blueprints_test.go (it may make sense to move that block of code and the global into a separate file soon, but I'll save that for later).